### PR TITLE
Support fewer data types in BlazorPackHubProtocol

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs [hunk 1]
```diff
@@ -445,10 +437,6 @@ private static object DeserializeObject(ref MessagePackReader reader, Type type,
                 {
                     return ReadString(ref reader, "argument");
                 }
-                else if (type == typeof(bool))
-                {
-                    return reader.ReadBoolean();
-                }
                 else if (type == typeof(int))
```
Comment (pranavkm) [inline src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs]: We don't actually need this, but we lose a lot of test coverage if we do so.

File: src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs [hunk 2]
```diff
@@ -457,17 +445,6 @@ private static object DeserializeObject(ref MessagePackReader reader, Type type,
                 {
                     return reader.ReadInt64();
                 }
-                else if (type == typeof(float))
-                {
-                    return reader.ReadSingle();
-                }
-                else if (type == typeof(byte[]))
```
Comment (pranavkm) [inline src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs]: Yaay

File: src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs [hunk 3]
```diff
@@ -420,18 +420,10 @@ private void SerializeArgument(ref MessagePackWriter writer, object argument)
...
 
-                case float floatValue:
-                    writer.Write(floatValue);
-                    break;
-
                 case byte[] byteArray:
                     writer.Write(byteArray);
                     break;
 
-                case Exception exception:
```
Comment (pranavkm) [inline src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs]: And gone

**Comment by davidfowl** [comment]
- No description
- No issue linked

What's the reason for doing this?

**Comment by rynowak** [comment]
I'm not sure why we'd do this other than removing exception (security)

**Comment by pranavkm** [comment]
These are data types not required for Blazor to function. Removing dead code particularly when there are security implications (serialization \ deserialization) seems like a good thing, no?

File: src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs [hunk 4]
```diff
@@ -457,17 +445,6 @@ private static object DeserializeObject(ref MessagePackReader reader, Type type,
                 {
                     return reader.ReadInt64();
                 }
-                else if (type == typeof(float))
```
Comment (mikaelm12) [inline src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs]: Sort of unrelated, why don't you use `is` for these checks?

Comment (davidfowl) [reply src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs]: It's a `Type` instance itself not an instance of an object.

Comment (pranavkm) [reply src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs]: I thought `is` only works on instances, such as `someInstance is float`, not when comparing two `Type` instances. Did I get that wrong?

Comment (mikaelm12) [reply src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs]: Ah, gotcha

**Comment by pranavkm** [comment]
*bump*

**Review by SteveSandersonMS — APPROVED** [review]
I'm happy with this. Thanks @pranavkm!

Before merging, could you check with Ryan whether he has strong opinions about removing the benign things like `float` etc? His comment suggests he's unconvinced this is the right thing to do.

For my purposes, I'm fine either way since we're not using these. I do lean towards removing stuff we're not using, except if this is going to come at significant maintenance cost if we're planning to "upgrade" to newer versions of MessagePack by copy/pasting code.

**Comment by rynowak** [comment]
I think it's wierd seeing these tests diverge for something that's harmless. We saw a security concern raised because `Exception` was in the list of types. There's no concern about primitives.

**Comment by pranavkm** [comment]
We already support fewer data types and consequently run a subset of the tests that SignalR's message pack does. Feels like a small extension to it

**Comment by pranavkm** [comment]
🆙 📅

**Review by rynowak — APPROVED** [review]

**Pull request merged**